### PR TITLE
sdk/java: accept multiple URLs in Client

### PR DESCRIPTION
**Pull request by jbowens**
Allow the Java SDK Client to be configured with multiple cored or
signerd URLs for high availability in the absence of a load balancer.

### CODE REVIEW

**Comment by kr** [comment]
:eyes:

File: sdk/java/src/main/java/com/chain/http/Client.java [hunk 1]
```diff
@@ -414,14 +455,15 @@ private Response checkError(Response response) throws ChainException {
...
-    } catch (URISyntaxException e) {
-      throw new MalformedURLException();
+  private void nextURL(int failedIndex) {
+    if (this.urls.size() == 1) {
+      return; // No point contending on the CAS if there's only one URL.
     }
+
+    // A request to the url at failedIndex just failed. Move to the next
+    // URL in the list.
+    int nextIndex = (failedIndex + 1) % this.urls.size();
```
Comment (kr) [inline sdk/java/src/main/java/com/chain/http/Client.java]: This looks like a data race on this.urls if addURL or setURL is called concurrently. Should the docs for those methods say they mustn't be called concurrently with any RPCs?

Comment (jbowens) [inline sdk/java/src/main/java/com/chain/http/Client.java]: Those methods are on the Builder whereas this one is on the Client. Once a Client is constructed, `urls` should be immutable.

**Comment by kr** [comment]
LGTM

**Comment by jeffomatic** [comment]
👀

File: sdk/java/src/main/java/com/chain/http/Client.java [hunk 2]
```diff
@@ -491,36 +532,55 @@ public Builder() {
...
+        throw new BadURLException(e.getMessage());
+      }
+      return this;
+    }
+
+    /**
+     * Adds a base URL for the client to use.
+     * @param url the URL of the Chain Core or HSM.
+     */
+    public Builder addURL(URL url) throws BadURLException {
```
Comment (jeffomatic) [inline sdk/java/src/main/java/com/chain/http/Client.java]: I don't think this method throws BadURLException.

**Comment by jbowens** [comment]
Tested it locally, and it worked okay, but I discovered #487.

**Pull request merged**